api/writeAPIBlocking.go

```diff
@@ -69,28 +70,28 @@ func NewWriteAPIBlockingWithBatching(org string, bucket string, service http2.Se
 }
 
 func (w *writeAPIBlocking) EnableBatching() {
-	w.mu.Lock()
-	defer w.mu.Unlock()
-	if !w.batching {
-		w.batching = true
+	if atomic.LoadInt32(&w.batching) == 0 {
+		w.mu.Lock()
+		w.batching = 1
 		w.batch = make([]string, 0, w.writeOptions.BatchSize())
+		w.mu.Unlock()
 	}
 }
 
 func (w *writeAPIBlocking) write(ctx context.Context, line string) error {
-	w.mu.Lock()
-	defer w.mu.Unlock()
-	body := line
-	if w.batching {
-		w.batch = append(w.batch, line)
-		if len(w.batch) == int(w.writeOptions.BatchSize()) {
-			body = strings.Join(w.batch, "\n")
-			w.batch = w.batch[:0]
-		} else {
-			return nil
-		}
+	if atomic.LoadInt32(&w.batching) > 0 {
+		return func(b string) error {
+			w.mu.Lock()
+			defer w.mu.Unlock()
+			w.batch = append(w.batch, b)
+			if len(w.batch) == int(w.writeOptions.BatchSize()) {
+				return w.flush(ctx)
+			} else {
+				return nil
+			}
+		}(line)
```

Comment:
Is there a reason to wrap the code into a function that is immediately invoked?

Reply:
The reason is the lock will be on during call flush and will be unlocked after this. The lock doesn't need to be on later.

Reply:
... there is IMHO no 'later' in the `write` function, the result of the function is directly returned

Reply:
You are right, the function doesn't need to be there. Even if the func wrapper is removed, both returns will be there, and unlock will be called immediately. I will remove it.